dashboard/app/models/user.rb

```diff
@@ -246,6 +247,24 @@ def delete_course_as_facilitator(course)
     courses_as_facilitator.find_by(course: course).try(:destroy)
   end
 
+  def log_admin_save
+    # Do not log for development, test, and adhoc environments.
+    return unless should_log?
+
+    ChatClient.message 'infra-security',
+      "#{admin ? 'Granting' : 'Revoking'} UserPermission: "\
+      "environment: #{rack_env}, "\
+      "user ID: #{id}, "\
+      "email: #{email}, "\
+      "permission: ADMIN",
+      color: 'yellow'
+  end
+
+  # don't log changes to admin permission in development, test, and ad_hoc environments
+  def should_log?
```

Comment:
I suggested extracting it into a method in this [comment](https://github.com/code-dot-org/code-dot-org/pull/16972#discussion_r132317861), so it can be stubbed in tests.

Even if we go with your 3rd suggestion (which I like), the check should be encapsulated in a method. 👍  to making it a class method.